src/reflect/value.go

```diff
@@ -641,30 +649,102 @@ func (v Value) OverflowFloat(x float64) bool {
 }
 
 func (v Value) MapKeys() []Value {
-	panic("unimplemented: (reflect.Value).MapKeys()")
+	if v.Kind() != Map {
+		panic(&ValueError{Method: "MapKeys", Kind: v.Kind()})
+	}
+
+	// empty map
+	if v.Len() == 0 {
+		return nil
+	}
+
+	var keys []Value
```

Comment:
This avoids a lot of allocations by doing a single allocation of the correct size:

```suggestion
	keys := make([]Value, 0, v.Len())
```